### PR TITLE
CBG-306: Function rename and removal of panic

### PR DESCRIPTION
**Pull request by JRascagneres**
- [x] Merge https://github.com/couchbaselabs/sync-gateway-accel/pull/233
- [x] Bump manifest

### CODE REVIEW

File: db/revision.go [hunk 1]
```diff
@@ -380,12 +384,12 @@ func containsUserSpecialProperties(body Body) bool {
 	return false
 }
 
-func canonicalEncoding(body Body) []byte {
+func canonicalEncoding(body Body) ([]byte, error) {
 	encoded, err := json.Marshal(body) //FIX: Use canonical JSON encoder
 	if err != nil {
-		panic(fmt.Sprintf("Couldn't encode body %v", body))
+		return nil, errors.New(fmt.Sprintf("Couldn't encode body %v", body))
```
Comment (bbrks) [inline db/revision.go:390]:
```suggestion
		return nil, fmt.Errorf("Couldn't encode body %v", body)
```

**Review by bbrks — CHANGES_REQUESTED** [review]

**Review by adamcfraser — COMMENTED** [review]
For channels.SetOf, we should:
- review non-test usage and switch to SetFromArray where appropriate (and handle the error)
- identify non-test use cases that aren't a good fit for SetFromArray, review how they can handle error
- for test usage, add a test-only API that logs a warning and returns an empty set on invalid channel names (for ease of use by tests).  Could potentially pass in testing.T and raise Fatalf on invalid channels to help clarify it's test-only.

File: db/document.go [hunk 2]
```diff
@@ -599,7 +599,8 @@ func (doc *document) updateChannels(newChannels base.Set) (changedChannels base.
 	}
 	if changed != nil {
 		base.Infof(base.KeyCRUD, "\tDoc %q / %q in channels %q", base.UD(doc.ID), doc.CurrentRev, base.UD(newChannels))
-		changedChannels = channels.SetOf(changed...)
+		changedChannels, err = channels.SetFromArray(changed, channels.KeepStar)
+		//panic("HAI")
```
Comment (bbrks) [inline db/document.go]: ohai

File: rest/blip_sync.go [hunk 3]
```diff
@@ -385,7 +385,12 @@ func (bh *blipHandler) sendChanges(sender *blip.Sender, params *subChangesParams
 
 	channelSet := bh.channels
 	if channelSet == nil {
-		channelSet = channels.SetOf(channels.AllChannelWildcard)
+		channelSetReturn, err := channels.SetFromArray([]string{channels.AllChannelWildcard}, channels.KeepStar)
+		if err != nil {
+			//panic("HAI")
```
Comment (bbrks) [inline rest/blip_sync.go]: ohai

**Review by bbrks — CHANGES_REQUESTED** [review]

File: channels/set.go [hunk 4]
```diff
@@ -66,22 +56,13 @@ func SetFromArray(names []string, mode StarMode) (base.Set, error) {
...
-		}
-	}
-	return nil
-}
-
 // Creates a set from zero or more inline string arguments.
 // Channel names must be valid, else the function will panic, so this should only be called
 // with hardcoded known-valid strings.
-func SetOf(names ...string) base.Set {
+func SetOf(tb testing.TB, names ...string) base.Set {
```
Comment (adamcfraser) [inline channels/set.go]: Move to a new util_testing.go inside channels package.

File: db/changes.go [hunk 5]
```diff
@@ -829,6 +829,7 @@ func (db *Database) getLateFeed(feedHandler *lateSequenceFeed) (<-chan *ChangeEn
 				Seq: logEntry.Sequence,
 			}
 			change := makeChangeEntry(logEntry, seqID, feedHandler.channelName)
+
```
Comment (adamcfraser) [inline db/changes.go]: Remove whitespace

File: rest/blip_sync.go [hunk 6]
```diff
@@ -385,7 +385,11 @@ func (bh *blipHandler) sendChanges(sender *blip.Sender, params *subChangesParams
 
 	channelSet := bh.channels
 	if channelSet == nil {
-		channelSet = channels.SetOf(channels.AllChannelWildcard)
+		channelSetReturn, err := channels.SetFromArray([]string{channels.AllChannelWildcard}, channels.KeepStar)
```
Comment (adamcfraser) [inline rest/blip_sync.go]: Can switch to base.SetOf and avoid error handling.

File: rest/changes_api.go [hunk 7]
```diff
@@ -249,7 +249,10 @@ func (h *handler) handleChanges() error {
 
 	// Get the channels as parameters to an imaginary "bychannel" filter.
 	// The default is all channels the user can access.
-	userChannels := ch.SetOf(ch.AllChannelWildcard)
+	userChannels, err := ch.SetFromArray([]string{ch.AllChannelWildcard}, ch.KeepStar)
```
Comment (adamcfraser) [inline rest/changes_api.go]: As above - base.SetOf

**Review by adamcfraser — CHANGES_REQUESTED** [review]
A few changes - the accel PR is merged and so can also bump the manifest.

**Review by adamcfraser — APPROVED** [review]
